sops.go

```diff
@@ -189,6 +190,53 @@ func (branch TreeBranch) Set(path []interface{}, value interface{}) TreeBranch {
 	return set(branch, path, value).(TreeBranch)
 }
 
+func unset(branch interface{}, path []interface{}) (interface{}, error) {
+	switch branch := branch.(type) {
+	case TreeBranch:
+		for i, item := range branch {
+			if item.Key == path[0] {
+				if len(path) == 1 {
+					branch = slices.Delete(branch, i, i+1)
+				} else {
+					v, err := unset(item.Value, path[1:])
+					if err != nil {
+						return nil, err
+					}
+					branch[i].Value = v
+				}
+				return branch, nil
+			}
+		}
+		return nil, fmt.Errorf("Key not found: %s", path[0])
```

Comment:
Instead of returning an error, one could also return the input unmodified (since the requested key is already gone). Then unset would be idempotent (same as setting a value).

But I guess this could also be made configurable (or added in a later PR).

Reply:
Agreed, I added a flag `--idempotent` to the `unset` command. If you have a better idea for the option name, please tell me :)
I want to avoid `--force` (like e.g. `rm`) because there is nothing to force here.
I decided to keep it as a flag, because when removing a key that should exist but is not there, users may want an error to be raised, instead of silently continuing.